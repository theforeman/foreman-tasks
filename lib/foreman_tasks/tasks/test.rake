namespace :test do
  task :foreman_tasks => 'db:test:prepare' do
    test_task = Rake::TestTask.new('foreman_tasks_test_task') do |t|
      t.libs << ['test', "#{ForemanTasks::Engine.root}/test"]
      t.test_files = ["#{ForemanTasks::Engine.root}/test/**/*_test.rb"]
      t.verbose = true
      t.warning = false
    end

    Rake::Task[test_task.name].invoke
  end
end

namespace :foreman_tasks do
  task :rubocop do
    begin
      require 'rubocop/rake_task'
      RuboCop::RakeTask.new(:rubocop_foreman_tasks) do |task|
	task.patterns = ["#{ForemanTasks::Engine.root}/app/**/*.rb",
		  "#{ForemanTasks::Engine.root}/lib/**/*.rb",
		  "#{ForemanTasks::Engine.root}/test/**/*.rb"]
      end
    rescue
      puts 'Rubocop not loaded.'
    end

    Rake::Task['rubocop_foreman_tasks'].invoke
  end
end

Rake::Task[:test].enhance do
  Rake::Task['test:foreman_tasks'].invoke
end

load 'tasks/jenkins.rake'
if Rake::Task.task_defined?(:'jenkins:unit')
  Rake::Task['jenkins:unit'].enhance ['foreman_tasks:rubocop', 'test:foreman_tasks']
end
