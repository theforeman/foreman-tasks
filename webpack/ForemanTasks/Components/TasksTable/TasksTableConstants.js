import { getControllerSearchProps } from 'foremanReact/constants';

export const TASKS_TABLE_ID = 'TASKS_TABLE';
export const SELECT_ROWS = 'SELECT_ROWS';
export const UNSELECT_ROWS = 'UNSELECT_ROWS';
export const UNSELECT_ALL_ROWS = 'UNSELECT_ALL_ROWS';
export const TASKS_RESUME_REQUEST = 'TASKS_RESUME_REQUEST';
export const TASKS_RESUME_SUCCESS = 'TASKS_RESUME_SUCCESS';
export const TASKS_RESUME_FAILURE = 'TASKS_RESUME_FAILURE';

export const CANCEL_CONFIRM_MODAL_ID = 'cancelConfirmModal';
export const RESUME_CONFIRM_MODAL_ID = 'resumeConfirmModal';
export const CANCEL_SELECTED_CONFIRM_MODAL_ID = 'cancelSelectedConfirmModal';
export const RESUME_SELECTED_CONFIRM_MODAL_ID = 'resumeSelectedConfirmModal';
export const UPDATE_CLICKED = 'UPDATE_CLICKED';

export const TASKS_SEARCH_PROPS = {
  ...getControllerSearchProps('tasks'),
  controller: 'foreman_tasks/tasks'
};
