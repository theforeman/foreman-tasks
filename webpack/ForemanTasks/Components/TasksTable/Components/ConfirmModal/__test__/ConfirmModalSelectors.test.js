import { testSelectorsSnapshotWithFixtures } from '@theforeman/test';
import {
  selectClicked,
  selectSelectedTasks,
  selectSelectedRowsLen,
} from '../ConfirmModalSelectors';
import { CANCEL_MODAL } from '../../../TasksTableConstants';

const state = {
  foremanTasks: {
    tasksTable: {
      tasksTableContent: {
        results: [
          {
            id: 1,
            action: 'action1',
            available_actions: { cancellable: true },
            can_edit: true,
          },
          { id: 2, action: 'action2', available_actions: { resumable: true } },
        ],
        itemCount: 10,
      },
      tasksTableQuery: {
        selectedRows: [1, 2, 3],
        clicked: { taskId: '1', taskName: 'test-task' },
        allRowsSelected: false,
      },
    },
  },
};

const fixtures = {
  'should select clicked': () => selectClicked(state),
  'should select selectedTasks': () => selectSelectedTasks(state),
  'should select selectedRowsLen 1': () =>
    selectSelectedRowsLen({
      ...state,
      foremanTasks: {
        tasksTable: {
          ...state.foremanTasks.tasksTable,
          tasksTableQuery: {
            ...state.foremanTasks.tasksTable.tasksTableQuery,
            clicked: { modalType: CANCEL_MODAL },
          },
        },
      },
    }),
  'should select selectedRowsLen all': () => selectSelectedRowsLen(state),
  'should select selectedRowsLen some': () =>
    selectSelectedRowsLen({
      ...state,
      foremanTasks: {
        tasksTable: {
          ...state.foremanTasks.tasksTable,
          tasksTableQuery: {
            ...state.foremanTasks.tasksTable.tasksTableQuery,
            allRowsSelected: true,
          },
        },
      },
    }),
};

describe('TasksDashboard - Selectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
