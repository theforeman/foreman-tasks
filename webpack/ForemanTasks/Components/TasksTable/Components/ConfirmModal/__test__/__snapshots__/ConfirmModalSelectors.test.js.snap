// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksDashboard - Selectors should select clicked 1`] = `
Object {
  "taskId": "1",
  "taskName": "test-task",
}
`;

exports[`TasksDashboard - Selectors should select selectedRowsLen 1 1`] = `3`;

exports[`TasksDashboard - Selectors should select selectedRowsLen all 1`] = `3`;

exports[`TasksDashboard - Selectors should select selectedRowsLen some 1`] = `0`;

exports[`TasksDashboard - Selectors should select selectedTasks 1`] = `
Array [
  Object {
    "canEdit": true,
    "id": 1,
    "isCancellable": true,
    "isResumable": undefined,
    "name": "action1",
  },
  Object {
    "canEdit": undefined,
    "id": 2,
    "isCancellable": undefined,
    "isResumable": true,
    "name": "action2",
  },
]
`;
